subprojects/wrapper/src/integTest/groovy/org/gradle/integtests/WrapperHttpsIntegrationTest.groovy

```diff
@@ -71,4 +75,51 @@ class WrapperHttpsIntegrationTest extends AbstractWrapperIntegrationSpec {
         then:
         outputDoesNotContain('WARNING Using HTTP Basic Authentication over an insecure connection to download the Gradle distribution. Please consider using HTTPS.')
     }
+
+    def "downloads wrapper via https proxy"() {
```

Comment:
❌ Missing test coverage: 
- "downloads wrapper via http proxy" (not https)
- proxy settings defined in an environmental variable
- proxy settings defined in an environmental variable and overridden by system properties

Reply:
>     * "downloads wrapper via http proxy" (not https)

That's covered by the existing `WrapperHttpIntegrationTest`.

>     * proxy settings defined in an environmental variable

I'm planning to add any missing tests as soon as I see the existing tests pass. Do you have any idea about the test failure I've [mentioned above](https://github.com/gradle/gradle/pull/12598#issuecomment-602071079)?

Reply:
_running build locally_

Reply:
It fails consistently on Linux and on MacOS as well. It seems to be a problem with the test itself (the proxy simply doesn't work). Running from the cmd line I get the same short error output, but the stack trace is available if I run the test from IDEA:
```
Using 8 worker leases.
Starting Build
Settings evaluated using settings file '/opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n/settings.gradle'.
Projects loaded. Root project using build file '/opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n/build.gradle'.
Included projects: [root project 'fai4n']

> Configure project :
Evaluating root project 'fai4n' using build file '/opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n/build.gradle'.
All projects evaluated.
Selected primary task 'wrapper' from project :
Tasks to be executed: [task ':wrapper']
Tasks that were excluded: []
:wrapper (Thread[Execution worker for ':',5,main]) started.

> Task :wrapper
Caching disabled for task ':wrapper' because:
  Build cache is disabled
Task ':wrapper' is not up-to-date because:
  No history is available.
:wrapper (Thread[Execution worker for ':',5,main]) completed. Took 0.052 secs.

BUILD SUCCESSFUL in 87ms
1 actionable task: 1 executed
Starting build with: /opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n/gradlew --no-daemon --stacktrace --gradle-user-home /opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n/user-home --warning-mode=all -Dmaven.repo.local=/opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n/m2-home-should-not-be-filled -Dorg.gradle.internal.repository.max.tentatives=1 hello
Working directory: /opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n
Environment vars:
    JAVA_HOME: /Applications/IntelliJ IDEA.app/Contents/jbr/Contents/Home
    GRADLE_HOME: 
    GRADLE_USER_HOME: null
    JAVA_OPTS: 
    GRADLE_OPTS: -Dorg.gradle.daemon.idletimeout=120000 "-Dorg.gradle.daemon.registry.base=/opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n/daemon" -Dorg.gradle.native.dir=/opt/Development/git/gradle/gradle/intTestHomeDir/worker-1/native -Dorg.gradle.deprecation.trace=true -Dorg.gradle.internal.reuse.user.home.services=false -Djava.io.tmpdir=/opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp -Dfile.encoding=UTF-8 -Dorg.gradle.classloaderscope.strict=true -Dorg.gradle.internal.launcher.welcomeMessageEnabled=false -ea -ea -Xms256m -Xmx512m -XX:MaxMetaspaceSize=512m -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/opt/Development/git/gradle/gradle/intTestHomeDir/worker-1
    GRADLE_RO_DEP_CACHE: 
Downloading http://127.0.0.1:53561/gradlew/dist

Exception in thread "main" java.io.IOException: Server returned HTTP response code: 407 for URL: http://127.0.0.1:53561/gradlew/dist
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1919)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1515)
	at org.gradle.wrapper.Download.downloadInternal(Download.java:149)
	at org.gradle.wrapper.Download.download(Download.java:128)
	at org.gradle.wrapper.Install$1.call(Install.java:68)
	at org.gradle.wrapper.Install$1.call(Install.java:48)
	at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:69)
	at org.gradle.wrapper.Install.createDist(Install.java:48)
	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:107)
	at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:63)

Gradle execution failed in /opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n with: /opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n/gradlew [--no-daemon, --stacktrace, --gradle-user-home, /opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n/user-home, --warning-mode=all, -Dmaven.repo.local=/opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n/m2-home-should-not-be-filled, -Dorg.gradle.internal.repository.max.tentatives=1, hello]
Process ExecResult:
{exitValue=1, failure=null}
-----
Output:
Downloading http://127.0.0.1:53561/gradlew/dist


-----
Error:
Exception in thread "main" java.io.IOException: Server returned HTTP response code: 407 for URL: http://127.0.0.1:53561/gradlew/dist
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1919)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1515)
	at org.gradle.wrapper.Download.downloadInternal(Download.java:149)
	at org.gradle.wrapper.Download.download(Download.java:128)
	at org.gradle.wrapper.Install$1.call(Install.java:68)
	at org.gradle.wrapper.Install$1.call(Install.java:48)
	at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:69)
	at org.gradle.wrapper.Install.createDist(Install.java:48)
	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:107)
	at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:63)

-----

org.gradle.integtests.fixtures.executer.UnexpectedBuildFailure: Gradle execution failed in /opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n with: /opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n/gradlew [--no-daemon, --stacktrace, --gradle-user-home, /opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n/user-home, --warning-mode=all, -Dmaven.repo.local=/opt/Development/git/gradle/gradle/subprojects/wrapper/build/tmp/test files/WrapperHttpIntegrationTest/downloads_wrapper_v...ted_proxy/fai4n/m2-home-should-not-be-filled, -Dorg.gradle.internal.repository.max.tentatives=1, hello]
Process ExecResult:
{exitValue=1, failure=null}
-----
Output:
Downloading http://127.0.0.1:53561/gradlew/dist


-----
Error:
Exception in thread "main" java.io.IOException: Server returned HTTP response code: 407 for URL: http://127.0.0.1:53561/gradlew/dist
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1919)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1515)
	at org.gradle.wrapper.Download.downloadInternal(Download.java:149)
	at org.gradle.wrapper.Download.download(Download.java:128)
	at org.gradle.wrapper.Install$1.call(Install.java:68)
	at org.gradle.wrapper.Install$1.call(Install.java:48)
	at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:69)
	at org.gradle.wrapper.Install.createDist(Install.java:48)
	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:107)
	at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:63)

-----

	at org.gradle.integtests.fixtures.executer.ForkingGradleHandle.unexpectedBuildStatus(ForkingGradleHandle.java:240)
	at org.gradle.integtests.fixtures.executer.ForkingGradleHandle.waitForStop(ForkingGradleHandle.java:222)
	at org.gradle.integtests.fixtures.executer.ForkingGradleHandle.waitForFinish(ForkingGradleHandle.java:190)
	at org.gradle.integtests.fixtures.executer.NoDaemonGradleExecuter.doRun(NoDaemonGradleExecuter.java:201)
	at org.gradle.integtests.fixtures.executer.AbstractGradleExecuter.run(AbstractGradleExecuter.java:1074)
	at org.gradle.integtests.fixtures.executer.AbstractDelegatingGradleExecuter.doRun(AbstractDelegatingGradleExecuter.java:29)
	at org.gradle.integtests.fixtures.executer.AbstractGradleExecuter.run(AbstractGradleExecuter.java:1074)
	at org.gradle.integtests.WrapperHttpIntegrationTest.downloads wrapper via authenticated proxy(WrapperHttpIntegrationTest.groovy:166)


```

Reply:
> It seems to be a problem with the test itself (the proxy simply doesn't work).

But aren't the existing proxy-related tests in `master` supposed to pass?

Reply:
What I meant is that your changes probably broke the tests. They work properly on master.

Reply:
They do pass. We'll need to investigate why your tests fail.

Reply:
They do work. I'm not sure why it fails for your case, it needs to be investigated.

Reply:
My guess is that `WrapperHttpsIntegrationTest` fails because the used [LittleProxy](https://github.com/adamfisk/LittleProxy) library does not support a HTTPS proxy.